state.go

```diff
@@ -21,6 +21,61 @@ import (
 	"github.com/pkg/errors"
 )
 
+// State transitions:
+//
+//                             init
+//
+//                               |        (wait timeout expired)
+//                               |   +---------------------------------+
+//                               |   |                                 |
+//                               v   v                                 |
+//                                         (auth req. failed)
+//                          bootstrapped ----------------------> authorize wait
+//
+//                                |
+//                                |
+//                                |  (auth data avail.)
+//                                |
+//                                v
+//
+//                           authorized
+//
+//          (update needs     |   |
+//           commit)          |   |
+//         +------------------+   |
+//         |                      |
+//         v                      |          (wait timeout expired)
+//                                |    +-----------------------------+
+//   update commit                |    |                             |
+//                                v    v                             |
+//         |                                (no update)
+//         +---------------> update check ---------------->  update check wait
+//
+//                                |
+//                                | (update ready)
+//                                v
+//
+//                           update fetch
+//
+//                                |
+//                                | (update fetched)
+//                                v
+//
+//                          update install
+//
+//                                |
+//                                | (update installed,
+//                                |  enabled)
+//                                |
+//                                v
+//
+//                              reboot
+//
+//                                |
+//                                v
+//
+//                              final (daemon exit)
```

Comment:
Hmm, I think actually systemd might respawn the daemon, even if only for a split second before the reboot goes through. Are we prepared for that?